### PR TITLE
Feature/fix nonormalize with azurepipeline

### PR DESCRIPTION
**Pull request by mtren**
## Related Issue
Maybe https://github.com/GitTools/GitVersion/issues/1967

## Motivation and Context

The combination of repos with detached head (like on buildservers) with the /nonormalize parameter did not work.

## How Has This Been Tested?

I testet it by executing gitversion within an azure devops buildserver pipeline by using the /nonormalize parameter.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/GitVersion.Core/Core/GitPreparer.cs [hunk 1]
```diff
@@ -68,7 +68,13 @@ private void PrepareInternal(bool normalizeGitDirectory, string? currentBranch,
             }
             else
             {
-                if (!normalizeGitDirectory) return;
+                if (!normalizeGitDirectory)
+                {
+                    if (currentBranch == null) return;
+                    EnsureLocalBranchExistsForCurrentBranch(EnsureOnlyOneRemoteIsDefined(), currentBranch);
```
Comment (asbjornu) [inline src/GitVersion.Core/Core/GitPreparer.cs:74]: It seems weird to me to perform normalization even to `/nonormalize` is being passed as an argument. Isn't it better to ensure that the Git repository is normalized before GitVersion is executed?

**Review by asbjornu — CHANGES_REQUESTED** [review]
Thanks for submitting this PR.

File: src/GitVersion.Core/Core/GitPreparer.cs (hunk 1 above)
Comment (mtren) [reply src/GitVersion.Core/Core/GitPreparer.cs:74]: GitVersion has low performance with many branches. I found out, that the normalization was the reason why it takes minutes.
But with the /nonormalize parameter GitVersion did not work when it is used on a buildserver.

Comment (asbjornu) [reply src/GitVersion.Core/Core/GitPreparer.cs:74]: `/nonormalize` works on build servers if you normalize the repository yourself. [GitVersion's operation requirements are found in the documentation](https://gitversion.net/docs/reference/requirements). When the repository is normalized to meet those requirements, it should execute successfully in any environment.

**Comment by arturcic** [comment]
Closing as it's not by design

**Pull request closed without merging**